libavcodec/qsvdec.c

```diff
@@ -851,6 +851,10 @@ static int qsv_decode(AVCodecContext *avctx, QSVContext *q,
             else
                 frame->flags &= ~AV_FRAME_FLAG_KEY;
         }
+        if (frame->width != outsurf->Info.CropW || frame->height != outsurf->Info.CropH) {
+            avctx->width  = frame->width  = outsurf->Info.CropW;
+            avctx->height = frame->height = outsurf->Info.CropH;
+        }
```

Comment:
widthxheight is the coded dimensions (in pixel) which is different from the cropping rectangle, see https://github.com/FFmpeg/FFmpeg/blob/master/libavutil/frame.h#L437-L446